core/src/main/java/com/scalar/db/api/Put.java

```diff
@@ -168,19 +184,303 @@ public Put withValue(String name, @Nullable ByteBuffer value) {
+   */
+  @Nullable
+  public ByteBuffer getBlobValueAsByteBuffer(String name) {
+    checkIfExists(name);
+
+    if (isNullValue(name)) {
+      // default value
+      return null;
+    }
+    assert values.get(name).isPresent();
+    return values.get(name).get().getAsByteBuffer().orElse(null);
+  }
+
+  /**
+   * Returns the BLOB value a byte array type for the specified name added to the list of put
+   * values.
+   *
+   * @param name a name
+   * @return the BLOB value for the specified name. If the value is NULL, null
+   */
+  @Nullable
+  public byte[] getBlobValueAsBytes(String name) {
+    checkIfExists(name);
+
+    if (isNullValue(name)) {
+      // default value
+      return null;
+    }
+    assert values.get(name).isPresent();
+    return values.get(name).get().getAsBytes().orElse(null);
+  }
+
+  /**
+   * Returns the value for the specified name added to the list of put values.
+   *
+   * <p>If the columns is a BOOLEAN type, it returns a {@code Boolean} object. If the columns is an
+   * INT type, it returns an {@code Integer} object. If the columns is a BIGINT type, it returns a
+   * {@code LONG} object. If the columns is a FLOAT type, it returns a {@code FLOAT} object. If the
+   * columns is a DOUBLE type, it returns a {@code DOUBLE} object. If the columns is a TEXT type, it
+   * returns a {@code String} object. If the columns is a BLOB type, it returns a {@code ByteBuffer}
+   * object.
+   *
+   * @param name a name
+   * @return the value for the specified name. If the value is NULL, null
+   */
+  @Nullable
+  public Object getValue(String name) {
+    checkIfExists(name);
+    if (isNullValue(name)) {
+      return null;
+    }
+
+    Optional<Value<?>> value = values.get(name);
+    assert value.isPresent();
+    if (value.get() instanceof BooleanValue) {
+      return getBooleanValue(name);
+    } else if (value.get() instanceof IntValue) {
+      return getIntValue(name);
+    } else if (value.get() instanceof BigIntValue) {
+      return getBigIntValue(name);
+    } else if (value.get() instanceof FloatValue) {
+      return getFloatValue(name);
+    } else if (value.get() instanceof DoubleValue) {
+      return getDoubleValue(name);
+    } else if (value.get() instanceof TextValue) {
+      return getTextValue(name);
+    } else if (value.get() instanceof BlobValue) {
+      return getBlobValue(name);
+    } else {
+      throw new AssertionError();
+    }
+  }
+
+  /**
+   * Indicates whether the list of put values contains the specified column.
+   *
+   * @param name a name
+   * @return whether the result contains the specified column name
+   */
+  public boolean containsColumn(String name) {
+    return values.containsKey(name);
+  }
+
+  /**
+   * Returns a set of the contained column names for the values added to the list of put values.
+   *
+   * @return a set of the contained column names
+   */
+  public Set<String> getContainedColumnNames() {
+    return ImmutableSet.copyOf(values.keySet());
+  }
```

Comment:
I also added these methods to get a value from `Put`. This methods assume to be used for tests.